src/App.tsx

```diff
@@ -16,36 +30,24 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortType>(SortType.INITIAL);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortedGoods = getGoodsInOrder(goodsFromServer, sortBy, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
-        </button>
-
-        <button type="button" className="button is-success is-light">
-          Sort by length
-        </button>
-
-        <button type="button" className="button is-warning is-light">
-          Reverse
-        </button>
+        <SortByAlphabet setSortBy={setSortBy} sortBy={sortBy} />
+        <SortByLength setSortBy={setSortBy} sortBy={sortBy} />
+        <ReverseList setIsReversed={setIsReversed} isReversed={isReversed} />
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(isReversed || sortBy !== SortType.INITIAL) && (
+          <ResetSort setSortBy={setSortBy} setIsReversed={setIsReversed} />
```

Comment:
Consider marking the active sorting button to improve user experience, as suggested in the checklist item 3. For example, you could conditionally apply a CSS class to highlight the active button.